### ISSUE TITLE
Google Image Search

### ISSUE THREAD

**Issue by RedDragonWebDesign** [yourduskquibbles/webannoyances#254]
I'm getting a floating header when I do Google Image Search.

https://www.google.com/search?q=table+of+c%2B%2B+and+java+collections&tbm=isch

![2020-10-08_180015](https://user-images.githubusercontent.com/43397938/95529760-abf78c80-0990-11eb-903f-df825dff403f.png)

I tried writing this filter, but it didn't like my combination of `:has()` and `:style()`

    google.com##header:has([aria-label="Search by image"]):style(position: static !important;)

This less specific one works, but there's a second problem of a large whitespace above it.

    google.com##header:style(position: static !important;)

High traffic website, so I figured I'd ask for help on this one.

**Comment by yourduskquibbles** [comment]
@RedDragonWebDesign Thanks for reporting! Are you able to test the filters below? if you watch closely on scrolldown/scrollup, there are lots of small changes the search bar makes which is why so many filters are necessary to keep the sizing consistent when going between scrolled down page and initial page.

I'm most concerned with how this could have unintended impact on the main google search function more-so than the google images since this is on the `www.google.com` domain and not limited to just image search.

```
www.google.com##header.Lj9fsd:style(position: absolute !important; top: 10px !important; box-shadow: none !important;)
www.google.com##.Lj9fsd.DU1Mzb .Si7Nvc:style(top: 0 !important;)
www.google.com##.Lj9fsd.DU1Mzb .F1hUFe:style(padding: 4px 28px 0 30px !important;)
www.google.com##.Lj9fsd.DU1Mzb .wQnou:style(height: 46px !important;)
www.google.com##.Lj9fsd.DU1Mzb .F7UV7d:style(height: 24px !important; width: 24px !important; background-size: 24px !important;)
www.google.com##.Lj9fsd.DU1Mzb .AWbbyf, .Lj9fsd.DU1Mzb .Aj0WOd, .Lj9fsd.DU1Mzb .lhkctf, .Lj9fsd.DU1Mzb .CzSCNb, .Lj9fsd.DU1Mzb .OWOjU, .Lj9fsd.DU1Mzb .AQxKeb:style(height: 24px !important; width: 24px !important;)
www.google.com##.Lj9fsd.DU1Mzb .JSAgYe:style(line-height: 34px !important;)
www.google.com##.Lj9fsd.DU1Mzb .zthhpf:style(height: 44px !important; align-items: normal !important;)
```

**Comment by yourduskquibbles** [comment]
@RedDragonWebDesign Have you had a chance to test this at all? I've had it in my own personal filters for about a week now and seeing no issues, thinking I will deploy soon unless you have noticed some issue on google search results that I haven't seen.

**Comment by RedDragonWebDesign** [comment]
Sorry for the delayed response. I'll throw these in my custom filters right now. I'll let you know if I notice anything.

**Comment by RedDragonWebDesign** [comment]
After scrolling down, there's some whitespace in the top right that might be worth fixing. Haven't noticed anything else.

![image](https://user-images.githubusercontent.com/43397938/98168159-ffc99880-1e9e-11eb-9dca-de8fca9fa909.png)

**Comment by yourduskquibbles** [comment]
@RedDragonWebDesign Awesome, thanks for feedback I will see if I can fix this and then push these filters out to Web Annoyances Ultralist.

**Comment by yourduskquibbles** [comment]
@RedDragonWebDesign can you confirm adding these two additional filters fixes the white space and makes the right pane full height of your browser?

`www.google.com##.l39u4d[style*="width:"]:style(top: 0 !important; height: 100% !important;)`
`www.google.com##.WaWKOe[style*="opacity:"]:style(height: 100% !important;)`


Full list of all filters to test:
```
www.google.com##.Lj9fsd.DU1Mzb .AWbbyf, .Lj9fsd.DU1Mzb .Aj0WOd, .Lj9fsd.DU1Mzb .lhkctf, .Lj9fsd.DU1Mzb .CzSCNb, .Lj9fsd.DU1Mzb .OWOjU, .Lj9fsd.DU1Mzb .AQxKeb:style(height: 24px !important; width: 24px !important;)
www.google.com##.Lj9fsd.DU1Mzb .F1hUFe:style(padding: 4px 28px 0 30px !important;)
www.google.com##.Lj9fsd.DU1Mzb .F7UV7d:style(height: 24px !important; width: 24px !important; background-size: 24px !important;)
www.google.com##.Lj9fsd.DU1Mzb .JSAgYe:style(line-height: 34px !important;)
www.google.com##.Lj9fsd.DU1Mzb .Si7Nvc:style(top: 0 !important;)
www.google.com##.Lj9fsd.DU1Mzb .wQnou:style(height: 46px !important;)
www.google.com##.Lj9fsd.DU1Mzb .zthhpf:style(height: 44px !important; align-items: normal !important;)
www.google.com##.WaWKOe[style*="opacity:"]:style(height: 100% !important;)
www.google.com##.l39u4d[style*="width:"]:style(top: 0 !important; height: 100% !important;)
www.google.com##header.Lj9fsd:style(position: absolute !important; top: 10px !important; box-shadow: none !important;)
```

**Comment by RedDragonWebDesign** [comment]
I did a quick test. Looks good to me!

**Comment by yourduskquibbles** [comment]
https://github.com/AdguardTeam/AdguardFilters/issues/67336#issuecomment-730471505

I think I will need to exclude these filters from android adguard mobile in the exclusion list but I don't have an Android device to test with, @RedDragonWebDesign any chance you have one to test with?

**Comment by RedDragonWebDesign** [comment]
Tested in Firefox For Android, with forced update of Web Annoyances Ultralist. Looks good.

I do see a problem in Windows 7+Chrome+1080p. If you click on an image, the vertical scroll bar for the right bar doesn't work.

Having 2 scroll bars is correct. But the left one is disabled. It should be scrollable, since there are useful page elements farther down (under where it says "Toolbox")

![image](https://user-images.githubusercontent.com/43397938/99964605-69272380-2d48-11eb-9dd9-4b717a7f485c.png)

**Comment by yourduskquibbles** [comment]
> 
> 
> Tested in Firefox For Android, with forced update of Web Annoyances Ultralist. Looks good.
> 
> I do see a problem in Windows 7+Chrome+1080p. If you click on an image, the vertical scroll bar for the right bar doesn't work.
> 
> Having 2 scroll bars is correct. But the left one is disabled. It should be scrollable, since there are useful page elements farther down (under where it says "Toolbox")
> 
> ![image](https://user-images.githubusercontent.com/43397938/99964605-69272380-2d48-11eb-9dd9-4b717a7f485c.png)

Thanks looks like others have reported this issue too, I will try to investigate fix for this today. Thanks for testing!

https://github.com/AdguardTeam/AdguardFilters/issues/68070#issuecomment-732807692

**Comment by yourduskquibbles** [comment]
@RedDragonWebDesign I think this should be fixed now, but leaving open in case you have time to confirm it is looking OK now. :)

**Comment by axeltrinidade** [comment]
Hi. Actually thought this[screenshot] was initially a bug due to the Google Search Fixer addon for FF android. After a bit of investigating, I found it to be caused by the Web Annoyances filter alone. It goes bonkers only on the image result page. Not faced an issue anywhere else.

![98811240-f362b400-2439-11eb-9b96-10b240b8d420](https://user-images.githubusercontent.com/68217806/100151904-4c5f2d00-2ebb-11eb-87e4-5bcc596e8603.jpg)

**Comment by RedDragonWebDesign** [comment]
@yourduskquibbles Looks good. Thanks for your time on this.

Looks like @axeltrinidade posted something in this thread too. Although that doesn't happen on my Firefox for Android. Alex, any idea what the difference is? Does that only happen when you also have your Google Search Fixer plugin turned on? Are you able to use the logger to narrow it down to a specific filter?

![2020-11-24 19 10 25](https://user-images.githubusercontent.com/43397938/100178514-da6df000-2e88-11eb-97b9-07b7b81f12cb.png)

**Comment by axeltrinidade** [comment]
Yes, it's a compatibility issue with Google Search Fixer. Not too much of a worry I think, because this annoyance has actually pushed me towards finally trying ddg!

**Comment by yourduskquibbles** [comment]
> Yes, it's a compatibility issue with Google Search Fixer. Not too much of a worry I think, because this annoyance has actually pushed me towards finally trying ddg!

@axeltrinidade So you confirm that if `Google Search Fixer` is disabled and uBO with Web Annoyances is only plugin enabled this does not happen? If no issue with only uBO I'm guessing there are some DOM element being removed or set to `display: none` by `Google Search Fixer` that `Web Annoyances` is also setting cosmetic properties on. It's possible if I can figure out what element is being set to not display by `Google Search Fixer` I could add the `display: block` property to the filter in Web Annoyances Ultralist and then it would be a race condition to see which plugin gets to set the property first.

Hard for me to really investigate that without an Android device to test with though and go looking through the Web Inspector

**Comment by axeltrinidade** [comment]
> So you confirm that if Google Search Fixer is disabled and uBO with Web Annoyances Ultralist is only plugin enabled this does not happen?

Yes.
Like I said, no worries or urgency. Maybe for the long run, would be better for the list to be more compatible.
